cmd/metal-api/internal/service/switch-service.go

```diff
@@ -325,6 +326,35 @@ func (r *switchResource) notifySwitch(request *restful.Request, response *restfu
 		}
 	}
 
+	if requestPayload.BGPPortStates != nil {
+		r.log.Debug("bgp port states", "id", id, "states", requestPayload.BGPPortStates)
+		for i, nic := range newSwitch.Nics {
+			bpsnew, ok := requestPayload.BGPPortStates[nic.Name]
+			if !ok && nic.BGPPortState == nil {
+				continue
+			}
+			if nic.BGPPortState == nil {
+				newSwitch.Nics[i].BGPPortState = nil
+				switchUpdated = true
+			}
+			if reflect.DeepEqual(nic.BGPPortState, &bpsnew) {
```

Comment:
can we detect equality by comparing members instead ? Using reflect.DeepEqual is not really recommended in production code